blocker/blocker.go

```diff
@@ -13,63 +12,49 @@ import (
 )
 
 const (
-	// blockBatchSize is the max number of (skylink) hashes to be sent for
+	// BlockBatchSize is the max number of (skylink) hashes to be sent for
 	// blocking simultaneously.
-	blockBatchSize = 100
-
-	// blockBatchSizeDivisor is the divisor applied to the batch size when an
-	// error is encountered.
-	blockBatchSizeDivisor = 10
-
-	// unableToUpdateBlocklistErrStr is a substring of the error returned by
-	// skyd if the blocklist was unable to get updated
-	unableToUpdateBlocklistErrStr = "unable to update the skynet blocklist"
+	BlockBatchSize = 100
 )
 
 var (
-	// retryInterval defines the amount of time between retries of blocked
-	// skylinks that failed to get blocked the first time around. This interval
-	// is (a lot) higher than the interval with which we scan for skylinks to
-	// get blocked.
-	retryInterval = build.Select(
-		build.Var{
-			Dev:      time.Minute,
-			Testing:  time.Second,
-			Standard: 4 * time.Hour,
-		},
-	).(time.Duration)
-
-	// sleepBetweenScans defines how long the scanner should sleep after
-	// scanning the DB and not finding any skylinks to scan.
-	sleepBetweenScans = build.Select(
+	// blockInterval defines the amount of time between fetching hashes that
+	// need to be blocked from the database.
+	blockInterval = build.Select(
 		build.Var{
 			Dev:      10 * time.Second,
 			Testing:  100 * time.Millisecond,
 			Standard: time.Minute,
 		},
 	).(time.Duration)
 
-	// sleepOnErrStep defines the base step for sleeping after encountering an
-	// error. We'll increase the sleep by an order of magnitude on each
-	// subsequent error until sleepOnErrSteps. We'll multiply that by the number
-	// of consecutive errors, up to sleepOnErrSteps times.
-	//
-	// Example: we'll sleep for 10 secs, then 20 and so on until 60. Then we'll
-	// keep sleeping for 60 seconds until the error is resolved.
-	sleepOnErrStep = 10 * time.Second
-	// sleepOnErrSteps is the maximum number of times we're going to increment
-	// the sleep-on-error length.
-	sleepOnErrSteps = 6
+	// retryInterval defines the amount of time between retries of blocked
+	// hashes that failed to get blocked the first time around. This interval
+	// is (a lot) higher than the blockInterval.
+	retryInterval = build.Select(
+		build.Var{
+			Dev:      time.Minute,
+			Testing:  time.Second,
+			Standard: time.Hour,
+		},
+	).(time.Duration)
 )
 
-// Blocker scans the database for skylinks that should be blocked and calls
-// skyd to block them.
-type Blocker struct {
-	staticCtx     context.Context
-	staticDB      *database.DB
-	staticLogger  *logrus.Logger
-	staticSkydAPI skyd.API
-}
+type (
+	// Blocker scans the database for skylinks that should be blocked and calls
+	// skyd to block them.
+	Blocker struct {
+		// latestBlockTime is the time at which we ran 'BlockHashes' the last
+		// time, this timestamp is used as an offset when fetch all 'new' hashes
+		// to block.
+		latestBlockTime time.Time
```

Comment:
That's not that much of an issue. On server restart we fetch the skyd blocklist and all hashes from the database, we diff them and we add what's missing. So this was intentional and not an issue IMO. I want to avoid persisting this value.